packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -327,3 +291,21 @@ function updateNamedCousins(rootNode, props) {
     }
   }
 }
+
+// In Chrome, assigning defaultValue to certain input types triggers input validation.
+// For number inputs, the display value loses trailing decimal points. For email inputs,
+// Chrome raises "The specified value <x> is not a valid email address".
+//
+// Here we check to see if the defaultValue has actually changed, avoiding these problems
+// when the user is inputting text
+//
+// https://github.com/facebook/react/issues/7253
+function synchronizeDefaultValue(node: Element, type: ?string, value: string) {
+  if (
+    // Focused number inputs synchronize on blur. See ChangeEventPlugin.js
+    (type !== 'number' || node.ownerDocument.activeElement !== node) &&
+    node.defaultValue !== value
+  ) {
+    node.defaultValue = value;
+  }
```

Comment:
Assigning `defaultValue` sets the `value` attribute, but doesn't have the side-effect of removing the password icon in IE. I'm not sure how I discovered this. I wish I had that kind of intuition for more useful things 😸.

<img width="493" alt="screen shot 2017-11-12 at 12 40 05 pm" src="https://user-images.githubusercontent.com/590904/32702140-3637ddf8-c7b0-11e7-8fe9-b4b0bc552980.png">